From 47f9f85c87e70969fce1d3b7862beaf211a62cad Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Thu, 22 May 2025 18:28:17 +0300
Subject: [PATCH] [JSW-31] jgs:removed ipm entries and rst entries

The buildroot external expects the macro names for
reset arc to be with _RESET_ instead of _RST_. Hence
using same macros in dtsi and enhanced the IPM driver
for FS1 and JGS (common) purpose.

Change-Id: I4159387eac900647d123bb5f5efb50c2ab254634
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/boards/intel/jgs/jgs_mgmt_asic_defconfig b/boards/intel/jgs/jgs_mgmt_asic_defconfig
index 39f9d4c..4e90996 100644
--- a/boards/intel/jgs/jgs_mgmt_asic_defconfig
+++ b/boards/intel/jgs/jgs_mgmt_asic_defconfig
@@ -20,9 +20,4 @@
 CONFIG_CLOCK_CONTROL_HL_PLL=y
 CONFIG_CLOCK_CONTROL_HL_PLL_V2=y
 CONFIG_HEAP_MEM_POOL_SIZE=4096
-
-CONFIG_IPM=y
-CONFIG_IPM_INTEL_FS=y
-CONFIG_IPM_INTEL_FS_TX_MBX=y
-CONFIG_IPM_INTEL_FS_RX_MBX=y
 CONFIG_HL_EFUSE=y
diff --git a/dts/arc/jgs_rst.dtsi b/dts/arc/jgs_rst.dtsi
index 66e8ff1..deb2067 100644
--- a/dts/arc/jgs_rst.dtsi
+++ b/dts/arc/jgs_rst.dtsi
@@ -13,15 +13,15 @@
 		zephyr,shell-uart = &uart0;
 	};
 
-	iccm0: iccm@ARC_RST_ICCM_OFFSET {
+	iccm0: iccm@ARC_RESET_ICCM_OFFSET {
 		device_type = "memory";
 		compatible = "arc,iccm";
-		reg = <PREFIX_0X(ARC_RST_ICCM_OFFSET) ARC_RST_ICCM_SIZE>;
+		reg = <PREFIX_0X(ARC_RESET_ICCM_OFFSET) ARC_RESET_ICCM_SIZE>;
 	};
 
-	dccm0: dccm@ARC_RST_DCCM_OFFSET {
+	dccm0: dccm@ARC_RESET_DCCM_OFFSET {
 		device_type = "memory";
 		compatible = "arc,dccm";
-		reg = <PREFIX_0X(ARC_RST_DCCM_OFFSET) ARC_RST_DCCM_SIZE>;
+		reg = <PREFIX_0X(ARC_RESET_DCCM_OFFSET) ARC_RESET_DCCM_SIZE>;
 	};
 };
